### PR TITLE
Separated CI jobs into several YAML files

### PR DESCRIPTION
**Pull request by Clovel**

### CODE REVIEW

**Comment by Clovel** [comment]
> This repository is not allowed to use macos build due to legal issues. Please remove it from this change.

Ok, I’ll remove it.

**Comment by Clovel** [comment]
> This repository is not allowed to use macos build due to legal issues. Please remove it from this change.

Done.

File: .github/workflows/linter.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Code checking
```
Comment (aajain-com) [inline .github/workflows/linter.yml:1]: nit; rename `Code checking` to `Lint Code` or something like that.

Comment (aajain-com) [reply .github/workflows/linter.yml:1]: Did you forget to commit your change?

Comment (Clovel) [reply .github/workflows/linter.yml]: Seems like the beginner mistake : forgot to push.

**Pull request merged**